rocketpy/Motor.py

```diff
@@ -0,0 +1,1937 @@
+            "constant",
+        )
+        return self.Kn
+
+    def evaluateInertia(self):
+        """Calculates propellant inertia I, relative to directions
+        perpendicular to the rocket body axis and its time derivative
+        as a function of time. Also calculates propellant inertia Z,
+        relative to the axial direction, and its time derivative as a
+        function of time. Products of inertia are assumed null due to
+        symmetry. The four functions are stored as an object of the
+        Function class.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        list of Functions
+            The first argument is the Function representing inertia I,
+            while the second argument is the Function representing
+            inertia Z.
+        """
+
+        # Inertia I
+        # Calculate inertia I for each grain
+        grainMass = self.mass / self.grainNumber
+        grainMassDot = self.massDot / self.grainNumber
+        grainNumber = self.grainNumber
+        grainInertiaI = grainMass * (
+            (1 / 4) * (self.grainOuterRadius**2 + self.grainInnerRadius**2)
+            + (1 / 12) * self.grainHeight**2
+        )
+
+        # Calculate each grain's distance d to propellant center of mass
+        initialValue = (grainNumber - 1) / 2
+        d = np.linspace(-initialValue, initialValue, grainNumber)
+        d = d * (self.grainInitialHeight + self.grainSeparation)
+
+        # Calculate inertia for all grains
+        self.inertiaI = grainNumber * grainInertiaI + grainMass * np.sum(d**2)
+        self.inertiaI.setOutputs("Propellant Inertia I (kg*m2)")
+
+        # Inertia I Dot
+        # Calculate each grain's inertia I dot
+        grainInertiaIDot = (
+            grainMassDot
+            * (
+                (1 / 4) * (self.grainOuterRadius**2 + self.grainInnerRadius**2)
+                + (1 / 12) * self.grainHeight**2
+            )
+            + grainMass
+            * ((1 / 2) * self.grainInnerRadius - (1 / 3) * self.grainHeight)
+            * self.burnRate
+        )
+
+        # Calculate inertia I dot for all grains
+        self.inertiaIDot = grainNumber * grainInertiaIDot + grainMassDot * np.sum(
+            d**2
+        )
+        self.inertiaIDot.setOutputs("Propellant Inertia I Dot (kg*m2/s)")
+
+        # Inertia Z
+        self.inertiaZ = (
+            (1 / 2.0)
+            * self.mass
+            * (self.grainOuterRadius**2 + self.grainInnerRadius**2)
+        )
+        self.inertiaZ.setOutputs("Propellant Inertia Z (kg*m2)")
+
+        # Inertia Z Dot
+        self.inertiaZDot = (1 / 2.0) * self.massDot * (
+            self.grainOuterRadius**2 + self.grainInnerRadius**2
+        ) + self.mass * self.grainInnerRadius * self.burnRate
+        self.inertiaZDot.setOutputs("Propellant Inertia Z Dot (kg*m2/s)")
+
+        return [self.inertiaI, self.inertiaZ]
+
+    def exportEng(self, fileName, motorName):
```

Comment:
Interesting... why was there a need to overwrite this method from the base class?

Reply:
Good catch! Can't see why these shouln't go on the base class, quite frankly.